### PR TITLE
Added Quick Sort in C

### LINKED ISSUE #2238: Addition of Quick Sort in C
## 🚀 Feature
Addition of quick sort in cp in C language.
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes.

## Motivation
Seeing as so many sorting techniques have been added before quick sort. It will be a valid and noteworthy addition to cp algorithms.

## Pitch
It is a crucial sorting technique and has been asked in many coding interviews.

### PR DESCRIPTION
**Pull request by shadowasphodel2919**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes, I have

### Description

I have added quick sort program to the sort category under C.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Solved the issue
https://github.com/TesseractCoding/NeoAlgo/issues/2238

### CODE REVIEW

File: C/sort/quickSort.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+/*Quick Sort is a divide and conquer algorithm which works on the basic principle
```
Comment (atarax665) [inline C/sort/quickSort.c:1]: Shift the description above the header file.

File: C/sort/quickSort.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            int temp = a[i];
+            a[i] = a[j];
+            a[j] = temp;
+        }
+    }
+    int temp = a[i + 1];
+    a[i + 1] = a[high];
+    a[high] = temp;
+    //returning the correct position of the index
+    return (i+1);
```
Comment (atarax665) [inline C/sort/quickSort.c]:
```suggestion
    return (i + 1);
```

File: C/sort/quickSort.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    for(i = 0; i <size; i++){
+        scanf("%d",&A[i]);
+    }
+
+    printf("\n");
+
+    quickSort(A, 0, size - 1);
+
+    printArray(A, size);
+}
```
Comment (atarax665) [inline C/sort/quickSort.c]: Add sample I/O and space and time complexity of the program in commented form.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by shadowasphodel2919** [comment]
@atarax665 I have made the required changes, kindly review again.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.geeksforgeeks.org/quicksort-using-random-pivoting/).

**Pull request closed without merging**